File: BubbleSortAlgorithm/Java/BubbleSort.java
package BubbleSortAlgorithm;

import java.util.Arrays;

public class BubbleSort {

    public static void main(String[] args) {
        int[] arr = {18, 12, 19, 18, 29, 89, 98};
        bubbleSort(arr);
        System.out.println("After Sorting the Integer Array: " + Arrays.toString(arr));
    }

    public static void bubbleSort(int[] arr) {
        boolean swap;
        for (int i = 0; i < arr.length - 1; i++) {
            swap = false;
            for (int j = 0; j < arr.length - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swap = true;
                }
            }
            if (!swap) {
                break;
            }
        }
    }
}

File: website/algorithms/bubble-sort-algorithm/BubbleSort.js
function bubbleSort(arr) {
  const n = arr.length;
  let swapped;

  do {
    swapped = false;

    for (let i = 0; i < n - 1; i++) {
      if (arr[i] > arr[i + 1]) {
        // Swap the elements
        const temp = arr[i];
        arr[i] = arr[i + 1];
        arr[i + 1] = temp;

        swapped = true;
      }
    }
  } while (swapped);

  return arr;
}

const unsortedArray = [97, 45, 5, 37, 10, 75, 19, 57, 85, 65, 24];
const sortedArray = bubbleSort(unsortedArray);
console.log(sortedArray);
